pallets/staking/src/lib.rs

```diff
@@ -3798,7 +3862,8 @@ where
         slash_fraction: &[Perbill],
         slash_session: SessionIndex,
     ) -> Result<Weight, ()> {
-        if !Self::can_report() {
+        // No need to run through when Slashing is off.
+        if !Self::can_report() || Self::slashing_status() == SlashingSwitch::Off {
```

Comment:
Perhaps the slashing switch check could be simplified to
```
!<SlashingStatus>::exists()
```
and the `Off` variant could be removed from the enum.

Reply:
If it's the storage value name that explains the semantics then it could be renamed to something like `SlashingAllowedFor`. Then when you say `!<SlashingAllowedFor>::exists()` it's clear that you mean that slashing is not allowed.